saleor/order/utils.py

```diff
@@ -585,6 +563,14 @@ def change_order_line_quantity(
             update_fields.extend(fields)
         else:
             line.save(update_fields=fields)
+
+        if catalogue_discount := line.discounts.filter(
```

Comment:
Why do you take this directly from DB instead of using `line_info.line_discounts`?

Reply:
Good point, my bad

Reply:
My bad, this id `OrderLineData` type:
```
@dataclass
class OrderLineData:
    variant_id: Optional[str] = None
    variant: Optional[ProductVariant] = None
    line_id: Optional[str] = None
    price_override: Optional[Decimal] = None
    quantity: int = 0
    rules_info: Optional[Iterable[VariantPromotionRuleInfo]] = None
```

Reply:
It was confusing so I added typing for orderLinesCreate and its utils :) Indeed it is `OrderLineInfo` type, however it only contains the line and quantity and it is constructed in the loop. The adding lines logic requires optimisation, to not create the lines in the loop, but using `bulk_create` instead. I cerated a ticket for that: https://linear.app/saleor/issue/SHOPX-1709